pep-0722.rst

```diff
@@ -221,57 +223,34 @@ Reference Implementation
 Code to implement this proposal in Python is fairly straightforward, so the
 reference implementation can be included here.
 
-A parser that reads *only* the script dependency metadata.
-
 .. code:: python
 
+   import re
    import tokenize
    from packaging.requirements import Requirement
 
-   DEPENDENCY_BLOCK_MARKER = "Script Dependencies:"
-
+   DEPENDENCY_BLOCK_MARKER = r"(?i)^#\s+script\s+dependencies:\s*$"
+   
    def read_dependency_block(filename):
-      # Use the tokenize module to handle any encoding declaration.
-      with tokenize.open(filename) as f:
-         for line in f:
-               if line.startswith("##"):
-                  line = line[2:].strip()
-                  if line == DEPENDENCY_BLOCK_MARKER:
-                     for line in f:
-                        if not line.startswith("##"):
-                              break
-                        line = line[2:].strip()
-                        if not line:
-                              continue
-                        # Try to convert to a requirement. This will raise
-                        # an error if the line is not a PEP 508 requirement
-                        yield Requirement(line)
-                     break
-
-A full metadata block parser that returns all metadata blocks in a script.
-
-.. code:: python
-
-   import tokenize
-   from packaging.requirements import Requirement
-
-   def read_metadata_blocks(filename):
-      # Use the tokenize module to handle any encoding declaration.
-      with tokenize.open(filename) as f:
-         for line in f:
-               if line.startswith("##"):
-                  block_type, sep, extra = line[2:].strip().partition(":")
-                  if not sep:
-                     continue
-                  block_data = []
-                  for line in f:
-                     if not line.startswith("##"):
+       # Use the tokenize module to handle any encoding declaration.
+       with tokenize.open(filename) as f:
+           for line in f:
+               if re.match(DEPENDENCY_BLOCK_MARKER, line):
+                   for line in f:
```

Comment:
@encukou left a comment about avoiding multiple-iter on the discuss thread -- would you agree to update the reference implementation to avoid this?

Reply:
I commented on Discourse, but basically no. I like the implementation as it is, it was easy to write and it's easy to follow. And to be honest, I'm not confident that I wouldn't make a mistake in a version that managed the "are we in a dependency block?" state manually - because I no longer *have* to do that sort of thing, thanks to Python's iterators. Maybe non-Python implementations won't be able to do this, but that's fine, they can implement their own version.